### ISSUE TITLE
[Feature Requests] professional header.thtml language attribute

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#269]
**Reported by @ivywe on 9 Dec 2008 01:18**

**Version:** 1.5.1

Description	professional header.thtml language attribute is not appear.
```HTML
'<html {lang_attribute}>' display as '<html >' but '<html lang="en">'.
```
[Mantis Bugtracker #782](http://project.geeklog.net/tracking/view.php?id=782)

[html_attribute.zip](https://github.com/Geeklog-Core/geeklog/files/89651/html_attribute.zip)

**Comment by mystralkk** [comment]
Related to #107

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 9 Dec 2008 06:56**

The language attribute is set only when multi-language support is enabled. Could that be the problem?

**Comment by mystralkk** [comment]
**Comment posted by Ivy on 9 Dec 2008 09:22**

It&#039;s important for web accessibility.

Web Content Accessibility Guidelines 1.0 ---- Priority 3
http://www.w3.org/TR/WAI-WEBCONTENT/

Checkpointsã4.3
http://www.w3.org/TR/WCAG10-HTML-TECHS/#identify-primary-lang

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 9 Dec 2008 09:41**

I see your point, but when multi-language support isn&#039;t enabled, Geeklog has no reliable way to provide that attribute. See bug #616 for details.

**Comment by mystralkk** [comment]
**Comment posted by Ivy on 10 Dec 2008 10:14**

If {lang_attribute} is set in all language files, Geeklog have always language attribute.
Now when multi-language support is enabled, if user choose non-supported language in &#039;My Account&#039;, Geeklog have no language attribute.

**Comment by mystralkk** [comment]
**Comment posted by mystral_kk on 2 Nov 2009 14:23**

We are testing a new way to decide {html_attribute} variable.

Step 1. When multi-language support is enabled, we just call COM_getLanguageId().

Step 2. If step #1 fails, we try to derive the language from $_USER[&#039;language&#039;] and converts it to a language id (e.g., &#039;en&#039;, &#039;ja&#039;, ...).

Step 3. If step #2 fails, we try to derive the language from $_CONF[&#039;locale&#039;].

Step 4. If step #3 fails (which is very unlikely), we use &#039;en&#039; as a default value.

I uploaded a PHP file.  Please have a look at it.

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 22 Jan 2011 18:48**

The problem with the patch is that it introduces a mapping table in a separate place (languageToLangId function). So it has to be updated when new languages are added.

I guess a better solution would be to add a proper ISO code to each language file, e.g.

$LANG_ISO639_1 = &#039;en&#039;;

and then use that for the header.

COM_getLanguageId would remain unchanged (it should always return an index for the $_CONF[&#039;language_files&#039;] array), but COM_setLangIdAndAttribute could then be changed to always use the new global variable and not call COM_getLanguageId any more.

(Actually, I couldn&#039;t figure out which ISO standard or RFC covers codes like de-DE or fr-CA, so $LANG_ISO639_1 is probably not a good name for that variable.)

**Comment by mystralkk** [comment]
Implemented with changeset fa9305b
